vllm/v1/spec_decode/eagle.py

```diff
@@ -997,20 +997,17 @@ def load_model(self, target_model: nn.Module) -> None:
                     "Target model does not have 'embed_tokens' or 'embedding' attribute"
                 )
 
-            # Check if shapes match and we found the embedding
-            eagle_shape = self.model.model.embed_tokens.weight.shape
-            target_shape = target_embed_tokens.weight.shape
-            if eagle_shape == target_shape:
+            if not self.model.has_own_embed_tokens:
                 logger.info(
-                    "Assuming the EAGLE head shares the same vocab embedding"
-                    " with the target model."
+                    "Draft model embed_tokens are uninitialized. "
```

Comment:
This was originally done as a memory optimization since the original EAGLE3 models are released with an embedding layer included, but having the same weights as the base model. 

Ideally we would have another check here to delete them if they are present but identical to those of the base model, but that's tricky to implement cleanly. I'm fine with doing it this way for now, but it should be noted that the behaviour is changing